clap_complete/src/aot/generator/mod.rs

```diff
@@ -61,9 +62,41 @@ pub trait Generator {
     ///     fn generate(&self, cmd: &Command, buf: &mut dyn Write) {
     ///         write!(buf, "{cmd}").unwrap();
     ///     }
+    /// #   fn try_generate(&self, cmd: &Command, buf: &mut dyn Write) -> Result<(), Error> {
+    /// #       write!(buf, "{cmd}")
+    /// #   }
+    /// }
+    /// ```
+    fn generate(&self, cmd: &Command, buf: &mut dyn Write) {
+        self.try_generate(cmd, buf)
+            .expect("failed to write completion file");
+    }
```

Comment:
>  For backwards compatibility make default trait implementation of generate function to call try_generate.

This needs to go the other way: `try_generate` needs to have an inherent function that calls `generate`